lib/cylc/rundb.py

```diff
@@ -315,47 +315,37 @@ def create_tables(self):
 
     def execute_queued_items(self):
         """Execute queued items for each table."""
-        will_retry = False
-        for table in self.tables.values():
-            # DELETE statements may have varying number of WHERE args
-            # so we can only executemany for each identical template statement.
-            for stmt, stmt_args_list in table.delete_queues.items():
-                self.connect()
-                if self._execute_stmt(table, stmt, stmt_args_list):
+        try:
+            for table in self.tables.values():
+                # DELETE statements may have varying number of WHERE args so we
+                # can only executemany for each identical template statement.
+                for stmt, stmt_args_list in table.delete_queues.items():
```

Comment:
No, delete statements must be done before insert statements. One use case is this: user cancels a broadcast, and immediately adds it back in. If we delete after insert, the broadcast entry will be lost.